pkg/antctl/raw/check/cluster/command.go

```diff
@@ -0,0 +1,210 @@
+			testContext.Fail("Test %s failed: %v", name, err)
+			numFailure++
+		} else {
+			testContext.Success("Test %s passed", name)
+			numSuccess++
+		}
+	}
+	testContext.Log("Test finished: %v tests succeeded, %v tests failed ", numSuccess, numFailure)
+	check.Teardown(ctx, testContext.client, testContext.clusterName, testContext.namespace)
+	if numFailure > 0 {
+		return fmt.Errorf("%v/%v tests failed", numFailure, len(testsRegistry))
+	}
+	return nil
+}
+
+func (t *testContext) setup(ctx context.Context) error {
+	t.Log("Creating Namespace %s for pre installation tests...", t.namespace)
+	_, err := t.client.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: t.namespace}}, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to create Namespace %s: %s", t.namespace, err)
+	}
+	deployment := check.NewDeployment(check.DeploymentParameters{
+		Name:        deploymentName,
+		Image:       getImageVersion(),
+		Replicas:    1,
+		Command:     []string{"bash", "-c"},
+		Args:        []string{"trap 'exit 0' SIGTERM; sleep infinity & pid=$!; wait $pid"},
+		Labels:      map[string]string{"app": "antrea", "component": "cluster-checker"},
+		HostNetwork: true,
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: "cni-conf", MountPath: "/etc/cni/net.d"},
+			{Name: "lib-modules", MountPath: "/lib/modules"},
+		},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:      "node-role.kubernetes.io/control-plane",
+				Operator: "Exists",
+				Effect:   "NoSchedule",
+			},
+			{
+				Key:      "node-role.kubernetes.io/master",
+				Operator: "Exists",
+				Effect:   "NoSchedule",
+			},
+			{
+				Key:      "node.kubernetes.io/not-ready",
+				Operator: "Exists",
+				Effect:   "NoSchedule",
+			},
+		},
+		Volumes: []corev1.Volume{
+			{
+				Name: "cni-conf",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/etc/cni/net.d",
+					},
+				},
+			},
+			{
+				Name: "lib-modules",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/lib/modules",
+						Type: ptr.To(corev1.HostPathType("Directory")),
+					},
+				},
+			},
+		},
+	})
+
+	t.Log("Creating Deployment")
+	_, err = t.client.AppsV1().Deployments(t.namespace).Create(ctx, deployment, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to create Deployment: %w", err)
+	}
+
+	t.Log("Waiting for Deployment to become ready")
+	err = check.WaitForDeploymentsReady(ctx, time.Second, podReadyTimeout, t.client, t.clusterName, t.namespace, deploymentName)
+	if err != nil {
+		return fmt.Errorf("error while waiting for Deployment to become ready: %w", err)
+	}
+	testPods, err := t.client.CoreV1().Pods(t.namespace).List(ctx, metav1.ListOptions{LabelSelector: "component=cluster-checker"})
+	if err != nil {
+		return fmt.Errorf("unable to list test Pod: %s", err)
+	}
+	if len(testPods.Items) > 0 {
+		t.testPod = &testPods.Items[0]
+	}
```

Comment:
even though in practice it should be fine because the deployment is ready, you should really return an error if you can't find a Pod, otherwise it affects tests down the line, and you could get a panic.

```
if len(testPods.Items) == 0 {
        return fmt.Errorf(...)
}
t.testPod = &testPods.Items[0]
return nil
```

Reply:
yes that sounds great